### PR TITLE
[meta] oci.Container component

### PR DESCRIPTION
**Pull request by PhilTaken**
Implement a basic oci container component

TODO: test with actual deployment

### CODE REVIEW

File: src/batou_ext/oci.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+@batou_ext.nix.rebuild
+class Container(Component):
+    """
+    A OCI Container component.
+    With this component you can dynamically schedule docker containers to be run on the target host
+
+    Example:
+    ```
+    self += batou_ext.oci.Container("mysql", "8.0")
```
Comment (frlan) [inline src/batou_ext/oci.py:22]: I think this example is not complete

File: src/batou_ext/oci.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    registry_address = Attribute(Optional[str], None)
+    registry_user = Attribute(Optional[str], None)
+    registry_password = Attribute(Optional[str], None)
+
+    _required_params_ = {
+        "image": "mysql",
+    }
+
+    def __init__(
```
Comment (zagy) [inline src/batou_ext/oci.py:41]: Interesting. We usually never override `__init__` but do everything in `configure()`

@ctheune What do you think about this?

File: src/batou_ext/oci.py (hunk 1 above)
Comment (zagy) [reply src/batou_ext/oci.py:23]: This works given the custom `__init__` below (which might be the cause of confusion)

File: src/batou_ext/oci.py (hunk 2 above)
Comment (ctheune) [reply src/batou_ext/oci.py:41]: This is highly unusual. I couldn't from the top of my head tell whether this breaks any assumptions about when this code gets executed. At least it does break all conventions of being able to access parent components etc. 

So, yes, this I think this should go into configure as well.

**Comment by PhilTaken** [comment]
hab die `__init__` logik einmal komplett nach `configure()` geschoben, mein probedeployment macht noch genau das gleiche, kein unterschied

**Pull request merged**